### ISSUE TITLE
We should introduce a hook to plug a different classDefinitionParser

### ISSUE THREAD

**Issue by Ducasse** [guillep/pharo-core#4653]
Since we want to make sure that we can have a new class definition and that other people can experiment freely with alternate classdefinition we need to able to ask a different classDefinitionParser (not well the term, here we are talking about the class definition and not the method definition parser). A potential solution is to introduce a classDefinitionParser method that should be sent to the superclass (now since the class will not be yet defined it look like we will need a kind of lookahead (on the superclass) and then reparse everything. 

It should be ok like that people will be able to write

```
MObject <<< # MyClass
    slots: { 
      (AttributeSlot named: #'x') type: #Integer; default: 2; beLazy.
      (AttributeSlot named: #'y') type: #Integer; default: 2; beLazy.
}
```
and we will able to have
```
Object subclass: (for backward)
```
and 
```
Object <<< # MyClass
     layout: #ephemerons;
     slots: {...}
```

**Comment by MarcusDenker** [comment]
yes, there is already the concept of #subclassDefinerClass, but that is a compiler (that is is does not only parse, it executes the class definition)

**Comment by Ducasse** [comment]
Ok I should finish first the fluid class parser definition

**Comment by Ducasse** [comment]
I think that we can close this issue. 
We have now the architecture to get multiple class definition printer.

**Comment by Ducasse** [comment]
For the compilation this is another issue and we should reopen an issue if needed.
